astropy/utils/shapes.py

```diff
@@ -25,6 +28,16 @@
     "unbroadcast",
 ]
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from types import EllipsisType
+    from typing import Literal, TypeVar
+
+    from numpy.typing import ArrayLike, NDArray
+    from typing_extensions import Self
+
+    DT = TypeVar("DT", bound=np.generic)
```

Comment:
Is there a better way of writing this down? I know about [`DTypeLike`](https://numpy.org/doc/stable/reference/typing.html#dtypelike), but that's for types that can be converted to a `dtype`, not for `dtype`s themselves.